### PR TITLE
feat: ab detection functionality

### LINKED ISSUE #2911: Implement proper ABs detection functionality for debugging
Based on the behaviour achieved at https://github.com/decentraland/unity-renderer/pull/2058 we should implement a proper functionality to toggle ABs/GLTFs material for debugging by using a chat command like `/detectABs` or `/detectABs scene`.

### Acceptance Criteria
- [ ] Be able to see every asset bundle with a green material and every GLTF with a red material when using the `/detectabs` chat command
- [ ] Be able to apply the same detection but only on the current scene by using the `/detectabsscene` chat command
- [ ] Be able to toggle the effect off

### PR DESCRIPTION
**Pull request by Antivortex**
## What does this PR change?

This PR:

1. Adds a tool for highlighting which mesh renderers are rendering the content loaded with asset bundles and which ones are loaded with raw GLTFs. DebugBridge receives a call from Kernel triggered by the chat command. This calls is passed to a ABDetectorPlugin via DataStore values as an intermediary. ABDetectorPlugin loads dedicated debug materials lazily on first call and applies them to the whole scene.
2. Adds custom low level data structure Multimap which is a generic wrapper for a dictionary of lists. After discussion with Alex we figured there are no class like that and decided to try using this one since it allows to not repeat a lot of code.

[This PR](https://github.com/decentraland/unity-renderer/pull/2058) was used as a reference for implementation.

Closes #2911

This Kernel PR implements chat commands for this feature:
https://github.com/decentraland/kernel/pull/536

New chat commands are:

/detectabs => enable for all shapes
/detectabs off => disable for all shapes
/detectabs scene => enable for current scene shapes
/detectabs scene off => disable for current scene shapes (works only if "/decetabs scene" was used for this scene to enable)

...

## How to test the changes?

This PR has to be tested together with the corresponding Kernel PR. To do this, use these links:

https://play.decentraland.zone/?renderer-branch=ab-detection-functionality&kernel-branch=ab-detection-functionality

1. Use chat command `/detectabs ` at some place with ABs enabled. The shapes should turn green.
2. Use chat command `/detectabs off` . The shapes should turn to their usual look.
3. Use chat command `/detectabs scene` . The shapes should turn green only within current scene.
4. Use chat command `/detectabs scene off` . The shapes of that last scene should turn to their usual look.
5. Command `/detectabs off` should also turn all shapes to normal look after  `/detectabs scene`

6. Repeat 1-2 with ABs disabled, the color should be red instead of green. To test with asset bundles disabled use this link:
https://play.decentraland.zone/?renderer-branch=ab-detection-functionality&kernel-branch=ab-detection-functionality&DISABLE_ASSET_BUNDLES=

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md

### CODE REVIEW

File: unity-renderer/Assets/Scripts/MainScripts/DCL/WorldRuntime/DebugController/DebugBridge.cs [hunk 1]
```diff
@@ -18,9 +18,16 @@ class ToggleSceneBoundingBoxesPayload
...
+        [Serializable]
+        class DetectABsPayload
+        {
+            public bool isOn;
+            public bool forCurrentScene;
+        }
 
         private ILogger debugLogger = new Logger(Debug.unityLogger.logHandler);
         private IDebugController debugController;
+        private double _lastTimeDetectABsCalled;
```
Comment (AjimenezDCL) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/WorldRuntime/DebugController/DebugBridge.cs:30]:
```suggestion
        private double lastTimeDetectABsCalled;
```

**Review by AjimenezDCL — CHANGES_REQUESTED** [review]

File: unity-renderer/Assets/DCLPlugins/EmoteAnimations/EmoteAnimationsPlugin.cs [hunk 2]
```diff
@@ -7,8 +7,16 @@ public class EmoteAnimationsPlugin : IPlugin
...
+            emotesAnimationTracker = new EmoteAnimationsTracker(DataStore.i.emotes, 
+                new EmoteAnimationLoaderFactory(), new WearableItemResolver(), 
+                Environment.i.serviceLocator.Get<IEmotesCatalogService>());
+        }
 
-        public void Dispose() { emotesAnimationTracker?.Dispose(); }
+        public void Dispose()
+        {
+            emotesAnimationTracker?.Dispose();
+        }
```
[the remark this answers is not in the document]
Comment (Antivortex) [reply unity-renderer/Assets/DCLPlugins/EmoteAnimations/EmoteAnimationsPlugin.cs:20]: @AjimenezDCL 
I am used to break long lines to improve readability by fitting them into the screen with Explorer tab opened, but it's not  necessery, we can revert this file change.

File: unity-renderer/Assets/Scripts/MainScripts/DCL/WorldRuntime/DebugController/DebugBridge.cs [hunk 3]
```diff
@@ -44,6 +51,25 @@ public void ShowFPSPanel()
             debugController.ShowFPSPanel();
         }
 
+        public void DetectABs(string payload)
+        {
+            var time = Time.timeSinceLevelLoad;
+            if (time - 1.0 < _lastTimeDetectABsCalled)
+            {
```
[the remark this answers is not in the document]
Comment (Antivortex) [reply unity-renderer/Assets/Scripts/MainScripts/DCL/WorldRuntime/DebugController/DebugBridge.cs:58]: You're right, this is redundant in here

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Helpers/Utils/Multimap.cs [hunk 4]
```diff
@@ -0,0 +1,139 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace DCL
+{
+    public class Multimap<KT, VT>
```
[the remark this answers is not in the document]
Comment (Antivortex) [reply unity-renderer/Assets/Scripts/MainScripts/DCL/Helpers/Utils/Multimap.cs:7]: Collection contains Keys property to return IEnumerable of keys and AllValues to return IEnumerable of all values to use LINQ on them. Do you suggest to implement IEnumerable<KeyValuePair<KT, List<VT>>> or something else?

File: unity-renderer/Assets/DCLPlugins/EmoteAnimations/EmoteAnimationsPlugin.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (AjimenezDCL) [reply unity-renderer/Assets/DCLPlugins/EmoteAnimations/EmoteAnimationsPlugin.cs]: Let's stick to what has been changed for the purpose of the PR. 

We usually dont break long lines, but I'm not against them. In any case, if you find a specific class hard to read, you can always create a small PR with these changes so we have better traceability of what's been changed and why without polluting the git log.

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Helpers/Utils/Multimap.cs (hunk 4 above)
[the remark this answers is not in the document]
Comment (AjimenezDCL) [reply unity-renderer/Assets/Scripts/MainScripts/DCL/Helpers/Utils/Multimap.cs:8]: that's probably enough, it is true that it wouldnt be much readable due to managing lists if we were to implement IEnumerable.

**Review by AjimenezDCL — APPROVED** [review]

**Pull request merged**